com.unity.template.renderstreaming-hd/Assets/Scripts/RenderStreaming.cs

```diff
@@ -196,19 +214,70 @@ void OnIceCandidate(ISignaling signaling, CandidateData e)
 
         void OnDataChannel(RTCPeerConnection pc, RTCDataChannel channel)
         {
-            Dictionary<int, RTCDataChannel> channels;
-            if (!mapChannels.TryGetValue(pc, out channels))
+            if (!mapChannels.TryGetValue(pc, out var channels))
             {
                 channels = new Dictionary<int, RTCDataChannel>();
                 mapChannels.Add(pc, channels);
             }
             channels.Add(channel.Id, channel);
 
-            if(channel.Label == "data")
+            if (channel.Label != "data")
             {
-                channel.OnMessage = new DelegateOnMessage(bytes => { RemoteInput.ProcessInput(bytes); });
-                channel.OnClose = new DelegateOnClose(() => { RemoteInput.Reset(); });
+                return;
             }
+
+            RemoteInput input = RemoteInputReceiver.Create();
+
+            // device.current must be changed after creating devices
+            m_defaultInput.MakeCurrent();
+
+            m_channelIdAndRemoteInput.Add(channel, input);
+            channel.OnMessage = bytes => m_channelIdAndRemoteInput[channel].ProcessInput(bytes);
+            channel.OnClose = () => OnCloseChannel(channel);
+
+            SimpleCameraController controller = m_listController
+                .FirstOrDefault(_controller => !m_remoteInputAndCameraController.ContainsValue(_controller));
+
+            if(controller != null)
+            {
+                controller.SetInput(input);
+                m_remoteInputAndCameraController.Add(input, controller);
+            }
+        }
+
+        void OnCloseChannel(RTCDataChannel channel)
+        {
+            RemoteInput input = m_channelIdAndRemoteInput[channel];
+            RemoteInputReceiver.Delete(input);
+
+            // device.current must be changed after removing devices
+            m_defaultInput.MakeCurrent();
+
+            if (m_remoteInputAndCameraController.TryGetValue(input, out var controller))
+            {
+                RemoteInput newInput = FindPrioritizedInput();
```

Comment:
Why are we adding newInput when removing input ?

Reply:
When a user disconnected, automatically assign another user the role to control a camera.

Reply:
I see. Ok, this code is a sample in the template, not in the package, so I think it should be fine.